From 1af7e853e252eaac9a6d664f3cc207bb9b40e1d4 Mon Sep 17 00:00:00 2001
From: Arian <arian.kulmer@web.de>
Date: Mon, 13 Mar 2023 15:46:13 +0100
Subject: [PATCH] ICustomCaptureResult: Add readout timestamp

commit [1] added two arguments to the CaptureResultExtras constructor.
Patch MiuiCamera to provide these arguments

https: //github.com/LineageOS/android_frameworks_base/commit/91e7522cb28d2b1e2e12adc8726db8190d711f2e
Change-Id: Iff64bf89566f6773e49496d29fe0099cda45bbf3
---
 .../protocol/ICustomCaptureResult.smali       | 32 +++++++++++++++++--
 1 file changed, 30 insertions(+), 2 deletions(-)

diff --git a/smali_classes2/com/xiaomi/protocol/ICustomCaptureResult.smali b/smali_classes2/com/xiaomi/protocol/ICustomCaptureResult.smali
index 56ecf5af8..c799f7fdc 100644
--- a/smali_classes2/com/xiaomi/protocol/ICustomCaptureResult.smali
+++ b/smali_classes2/com/xiaomi/protocol/ICustomCaptureResult.smali
@@ -346,7 +346,7 @@
 .end method
 
 .method public static toTotalCaptureResult(Lcom/xiaomi/protocol/ICustomCaptureResult;IZ)Landroid/hardware/camera2/TotalCaptureResult;
-    .locals 23
+    .locals 27
 
     const-string v0, "android.hardware.camera2.impl.PhysicalCaptureResultInfo"
 
@@ -434,7 +434,7 @@
 
     if-lt v7, v8, :cond_0
 
-    const/16 v7, 0xb
+    const/16 v7, 0xd
 
     :try_start_1
     new-array v8, v7, [Ljava/lang/Class;
@@ -488,6 +488,18 @@
 
     aput-object v19, v8, v21
 
+    sget-object v19, Ljava/lang/Boolean;->TYPE:Ljava/lang/Class;
+
+    const/16 v23, 0xb
+
+    aput-object v19, v8, v23
+
+    sget-object v19, Ljava/lang/Long;->TYPE:Ljava/lang/Class;
+
+    const/16 v24, 0xc
+
+    aput-object v19, v8, v24
+
     .line 8
     invoke-virtual {v1, v8}, Ljava/lang/Class;->getDeclaredConstructor([Ljava/lang/Class;)Ljava/lang/reflect/Constructor;
 
@@ -565,6 +577,22 @@
 
     aput-object v19, v7, v21
 
+    const/16 v25, 0x0
+
+    invoke-static/range {v25 .. v25}, Ljava/lang/Boolean;->valueOf(Z)Ljava/lang/Boolean;
+
+    move-result-object v19
+
+    aput-object v19, v7, v23
+
+    const/16 v26, 0x0
+
+    invoke-static/range {v26 .. v26}, Ljava/lang/Integer;->valueOf(I)Ljava/lang/Integer;
+
+    move-result-object v19
+
+    aput-object v19, v7, v24
+
     .line 11
     invoke-virtual {v8, v7}, Ljava/lang/reflect/Constructor;->newInstance([Ljava/lang/Object;)Ljava/lang/Object;
 
-- 
GitLab
